docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,237 @@ The remaining points can happen any time after 1) is complete although note that
+  be able to use `cylc clean` to clean out the log directory (and whatever other
+  directories you want removing) after which `cylc play` can be used.
+  Note that this means that `cylc play` must be able to use the command
+  line options from the previous run as stored in the private database.
+  There are 2 options:
+  * Require the use of a `--re-run` option to enable this
+    (otherwise `cylc play` will fail if it finds an existing private database).
+  * Do this automatically if no existing `log` directory is found.
+* Note that `cylc play` will always load the latest workflow definition found in
+  the run directory and will respect any command line options (which may alter
+  the workflow definition). Therefore, continuing a workflow using `cylc play`
+  effectively implies a reload.
+  * The processed `cylc.flow` files will be kept so there will be a record of
+    any changes
+    (see also [cylc-flow #3763](https://github.com/cylc/cylc-flow/issues/3763)).
+* Support Rose optional configuration via the environment variable
+  `ROSE_SUITE_OPT_CONF_KEYS` or the option `--rose-opt-conf-key=KEY`.
+
+`cylc reinstall` - new command to re-install workflows.
+* Very similar to the `cylc install` command except that this applies to a
+  previously installed workflow (i.e. you run this command in the directory of
+  the installed workflow, not in the source directory).
+* The rsync will use `--delete` to ensure that any previously installed files
+  which have been removed from the source directory also get removed from the
+  installed workflow.
+  * Note that this will have the effect of removing any files installed via
+    `rose-suite.conf`. They will get recreated by a subsequent reload or play but
+    could affect a running workflow.
+* Any options specified as part of the original install will be retained unless
+  overridden on the command line.
+* Consider a `--dry-run` option to report what would be changed?
+  * Does this replace `rose suite-cmp-vc`?
+
+`cylc reload` - no change from existing command.
+
+`cylc validate` - as per existing command with the following changes:
+* If validating a workflow installed with `cylc install` then use any relevant
+  command line options specified as part of the install unless overridden on the
+  command line.
+* Will work fine on source directories which include `rose-suite.conf` files.
+* No `--strict` mode
+  (naked dummy tasks no longer allowed - all tasks must be declared).
+  * This one is controversial and requires further discussion!
+* Cyclic graph validation can be quite slow and is currently included as part of
+  `--strict`. Add a `--no-cyclic-graph-validation` option to disable this?
+
+Several combinations of the above commands will be commonly used.
+Propose to support these as separate commands:
+* `cylc install-play` (`cylc ip` for short?)
+* `cylc install-validate`
+* `cylc reinstall-reload`
+
+`cylc pause|unpause` - new commands replacing `cylc hold|release|unhold`
```

Comment:
@dpmatthews is not so fond of this command combination.

I'm rather fond of keeping functionalities separate but admit-ably do appreciate the simplicity of the tape-deck model and the removal of another major command.

Reply:
> the removal of another major command.

Let's do it then. We have too many commands!

Reply:
I'm confused. Are you suggesting we use the play command to unpause tasks?

Reply:
Yes, that's what I'm suggesting, unless there's a downside beyond what I've suggested in the first comment above (or you think that is in itself sufficiently bad).

Reply:
The interface for `cylc release` is completely different, e.g.:
 cylc release REG TASK_GLOB ... - release one or more tasks
It doesn't make sense to me to make this part of `cylc play`.

Reply:
I agree `unpause` is ugly but I'm not sure what else to suggest.
I'd be happy to stick with hold/release, especially since we still plan to use `held` rather than `paused` as a task attribute:
https://cylc.github.io/cylc-admin/proposal-state-names.html#new-abstract-task-states

Reply:
Sorry, you're right, I forgot about holding individual tasks. Although we still need to figure out how to do that under SoD  https://github.com/cylc/cylc-flow/issues/3743

Reply:
Once the held state is implemented via xtriggers we will require a `set|satisfy xtrigger` interface (which would be quite nice, say for adding/removing clock triggers via the GUI). So holding a task becomes a special case of that.

* play/pause for suite state, hold/release for task state?
* pause/release for both?

Reply:
... or hold/release for both? (i.e. no change)

Reply:
If we're going for the tape deck analogy, we might as well be consistent, which suggests we need play/pause at least as an alias.  As per Oliver's comment, play/pause makes sense for the flow as a whole, but not really for individual tasks.

So my vote is:
- play/pause for flow state, with aliases: play=release=unpause (if paused) and pause=hold
- hold/release for task state, with aliases: hold=pause and release=unpause

(The aliases are all pretty obvious and easily understood, but will help out users if the flow vs task distinction bleeds together a bit in the minds...)

Reply:
I've updated the proposal to add the aliases.
The use of `cylc play` to release a paused workflow is already noted as an option.
Could we get this merged and make a decision on that later?

Reply:
Ok, add a TODO note or something to mark this apart from the more solid parts of the proposal.

Reply:
Done